src/decoratedtextinput/DecoratedTextInput.ts

```diff
@@ -0,0 +1,33 @@
+import { DNode } from '@dojo/widget-core/interfaces';
+import { theme } from '@dojo/widget-core/mixins/Themed';
+import { v } from '@dojo/widget-core/d';
+
+import TextInput, { TextInputProperties } from '../textinput/TextInput';
+import * as css from './styles/decoratedtextinput.m.css';
+
+export interface DecoratedTextInputProperties extends TextInputProperties {
+	addonAfter: DNode[];
+	addonBefore: DNode[];
+}
+
+@theme(css)
+export default class DecoratedTextInput extends TextInput<DecoratedTextInputProperties> {
```

Comment:
I hate to be that guy, but the name `DecoratedTextInput` feels weird to me since we're overloading the term "decorated" in the context of ES.

Reply:
I changed it to `EnhancedTextInput`. DEBATE AWAY